fvm/environment/event_emitter.go

```diff
@@ -215,6 +235,37 @@ func (emitter *eventEmitter) EmitEvent(event cadence.Event) error {
 	return nil
 }
 
+func (emitter *eventEmitter) EmitFlowEvent(etype flow.EventType, payload []byte) error {
+	defer emitter.tracer.StartExtensiveTracingChildSpan(
+		trace.FVMEnvEmitEvent).End()
+
+	err := emitter.meter.MeterComputation(ComputationKindEmitEvent, 1)
+	if err != nil {
+		return fmt.Errorf("emit event failed: %w", err)
+	}
+
+	payloadSize := uint64(len(payload))
+
+	flowEvent := flow.Event{
+		Type:             etype,
+		TransactionID:    emitter.txID,
+		TransactionIndex: emitter.txIndex,
+		EventIndex:       emitter.eventCollection.TotalEventCounter(),
+		Payload:          payload,
+	}
+
+	// TODO: to set limit to maximum when it is service account and get rid of this flag
+	isServiceAccount := emitter.payer == emitter.chain.ServiceAddress()
```

Comment:
Is it okey if I remove this? I was also assuming the limits are adjusted when the transaction is validated and we let service account define any limit instead of whitelisiting when checking ?

Reply:
The service transaction won't be using these particular events (right?) so it doesn't need an exception. If an exception is needed, I would rather add it during fvm `Config` construction by increasing the limit.

Reply:
right, at least for now we don't need to support it. going to remove this condition.